channels/facebook/multiple_post/index.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright IBM Corp. 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const openwhisk = require('openwhisk');
+
+/**
+ * Examines the reply from Conversation and posts one or more replies to the channel as needed.
+ *
+ * @param  {JSON} params - output JSON to be posted back to the channel
+ * @return {JSON}        - result of the call(s) to the Post action
+ */
+function main(params) {
+  return new Promise(resolve => {
+    resolve(postMultipleMessages(params));
+  });
+}
+
+/**
+ * Checks the length of Conversation's message.text array and sends one
+ * reply to the channel for each element in the
+ * array.
+ *
+ * @param  {JSON} params - output JSON to be posted back to the channel
+ * @return {JSON}        - result of the call(s) to the Post action
+ */
+function postMultipleMessages(params) {
+  // Determine the full path of the channel's postsequence sequence
+  const packageName = extractCurrentPackageName(process.env.__OW_ACTION_NAME);
+  const deployName = packageName.split('_')[0];
+  const sequenceName = `${deployName}_postsequence`;
+
+  // At minimum we will send one message to the channel,
+  // but if Conversation contains an array, send more
+  const size = Array.isArray(params.message) ? params.message.length : 1;
+
+  const responses = [];
+
+  // Post the message(s) and ultimately return JSON
+  // with one response for each channel post that occurred.
+  return postMessage(sequenceName, params, responses, size, 0).then(() => {
+    return { postResponses: responses };
+  });
+}
+
+/**
+ * Recursively calls postMessage until all Conversation messages have been sent to the channel.
+ *
+ * @param sequenceName name of channel's postsequence sequence
+ * @param params the channel specific params to call the channel's api
+ * @param responses an array to store the response in case we have to build it over multiple posts
+ * @param size the number of messages to send to the channel
+ * @param index the array index of the message that needs to be sent
+ * @returns {*}
+ */
+function postMessage(sequenceName, params, responses, size, index) {
```

Comment:
here's a common way i've seen that shorthands and/or optimizes recursive promise calls (taken from [here](https://stackoverflow.com/questions/24586110/resolve-promises-one-after-another-i-e-in-sequence)):
```javascript
const sequence = Promise.resolve();

return facebook.messages
  .reduce((seq, message) => {
    seq
      .then(result => {
        return performAction(message);
      })
      .catch(error => {
        throw error;
      })
  }, sequence);
```
The Promise.resolve() `sequence` is the start of the promise chain, and every the chain resolves, the reducer will return and pass along the next promise to be called. This should make it easier to read, without worrying about array indices and the when-to-break-out-of-recursive-function business.

Reply:
appreciate you teaching me this, I'm not going to hold up this current merge to develop as I don't want to slow things down but I hope to try this out in a future PR.